src/test/repository/EntityIntegration.test.ts

```diff
@@ -928,4 +932,60 @@ describe('Integration tests for all entities', function () {
       fail('Should not have failed: ' + e);
     }
   });
+
+  test('Promotions should tell whether user has voted the promotion or not if userId is provided to query', async () => {
+    const user = new UserFactory().generate();
+    const promotion1 = new PromotionFactory().generateWithRelatedEntities(user);
+    const promotion2 = new PromotionFactory().generateWithRelatedEntities(user);
+    const promotion3 = new PromotionFactory().generateWithRelatedEntities(user);
+    await userRepository.save(user);
+    await promotionRepository.save(promotion1);
+    await promotionRepository.save(promotion2);
+    await promotionRepository.save(promotion3);
+    // promotion1
+    let voteRecord: VoteRecord = new VoteRecord(user, promotion1);
+    await voteRecordRepository.save(voteRecord);
+    voteRecord.voteState = VoteState.UP;
+    await promotionRepository.increment({ id: promotion1.id }, 'votes', 1);
+    await voteRecordRepository.save(voteRecord);
+    // promotion2
+    voteRecord = new VoteRecord(user, promotion2);
+    await voteRecordRepository.save(voteRecord);
+    voteRecord.voteState = VoteState.DOWN;
+    await promotionRepository.decrement({ id: promotion2.id }, 'votes', 1);
+    await voteRecordRepository.save(voteRecord);
+
+    const promotionQueryDTO: PromotionQueryDTO = {
+      userId: user.id,
+    };
+
+    try {
+      const promotions: Promotion[] = await promotionRepository.getAllPromotions(
+        promotionQueryDTO
+      );
+
+      expect(promotions).toBeDefined();
+      expect(promotions.length).toEqual(3);
+      expect(
+        Number(
+          promotions.find((promotion) => promotion.id === promotion1.id)
+            ?.voteState
+        )
+      ).toEqual(1);
+      expect(
+        Number(
+          promotions.find((promotion) => promotion.id === promotion2.id)
+            ?.voteState
+        )
+      ).toEqual(VoteState.DOWN);
+      expect(
+        Number(
+          promotions.find((promotion) => promotion.id === promotion3.id)
+            ?.voteState
+        )
```

Comment:
I don't think we need the `Number(...)` part, we should be able to compare using `voteState`

Reply:
The reason why I put Number is there's weird error that the voteState becomes `string` not number when compare. But , there's no such error anymore, so yeah I gonna remove Number.